pennylane/devices/default_qubit_torch.py

```diff
@@ -239,9 +239,12 @@ def _asarray(self, a, dtype=None):
             if not isinstance(a[0], torch.Tensor):
                 res = np.asarray(a)
                 res = torch.from_numpy(res)
+                res = torch.cat([torch.reshape(i, (-1,)) for i in res], dim=0)
+            elif len(a) == 1 and len(a[0].shape) > 1:
+                res = torch.cat(a, dim=0)
```

Comment:
I tried this solution but unfortunately it is still unrolling the density matrix, if I don't use `torch.cat(a, dim=0)` I also run into an error. But I think the logic works well because it is only applied when there is only one measurement and therefore it should be safe. Another idea @antalszava ?

Reply:
With the previous logic you are doing `[tensor([[0.5000+0.j, 0.5000+0.j],
        [0.5000+0.j, 0.5000+0.j]], dtype=torch.complex128)]` ->
`tensor([0.5000+0.j, 0.5000+0.j, 0.5000+0.j, 0.5000+0.j],
       dtype=torch.complex128)` ->
`tensor([0.5000+0.j, 0.5000+0.j, 0.5000+0.j, 0.5000+0.j],
       dtype=torch.complex128)`

Reply:
with the new logic:
`a=[tensor([[0.5000+0.j, 0.5000+0.j],
        [0.5000+0.j, 0.5000+0.j]], dtype=torch.complex128)]`
->
`res=tensor([[0.5000+0.j, 0.5000+0.j],
        [0.5000+0.j, 0.5000+0.j]], dtype=torch.complex128)`

Reply:
Oh, I see! Then a minor suggestion: how about doing `res=a[0]`? Concatenation seems a bit of an overkill to achieve that.

Reply:
I changed it 👍